lib/deaf_grandma.rb

```diff
@@ -19,8 +24,19 @@ def run!
 
 
   def speak(input)
+    if input.upcase == input && input == 'BYE'
+      @bye_counter = @bye_counter + 1
+      if @bye_counter == 3
```

Comment:
The duplication of `if @bye_counter == 3` here and in the run loop is unfortunate. Minimally, I suggest extracting the `3` into a constant. But I'd rather see `@bye_counter == 3` extracted into a new method. Maybe call it `exiting?` or `time_to_exit?`.